### ISSUE TITLE
[Bug] convert yolox to onnx, isolated identity node

### ISSUE THREAD

**Issue by shuyuan-wang** [open-mmlab/mmdeploy#1712]
### Checklist

- [X] I have searched related issues but cannot get the expected help.
- [x] 2. I have read the [FAQ documentation](https://github.com/open-mmlab/mmdeploy/blob/master/docs/en/faq.md) but cannot get the expected help.
- [x] 3. The bug has not been fixed in the latest version.

### Describe the bug

When I try to convert yolox_s model to onnx, although the test result is correct, there are several identity node in onnx file which doesn't seem right

The converted onnx is in the following link, please take a look.
链接: https://pan.baidu.com/s/1APp-pQNsI9bTgzo-knbQ3Q?pwd=kj5p 提取码: kj5p 

### Reproduction

`python tools/deploy.py mmdeploy/configs/mmdet/detection/yolox_onnx.py mmdetection/configs/yolox/yolox_s_8x8_300e_coco.py yolox_s_8x8_300e_coco_20211121_095711-4592a793.pth mmdeploy/demo/resources/det.jpg --work-dir mmdeploy/yolox_work_dir --test-img mmdeploy/demo/resources/det.jpg`

### Environment

```Shell
- GCC 9.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.6.0 (Git Hash 52b5f107dd9cf10910aaa19cb47f3abf9b349815)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.6
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
  - CuDNN 8.3.2  (built against CUDA 11.5)
  - Magma 2.6.1
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.6, CUDNN_VERSION=8.3.2, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.12.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=OFF, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 

2023-02-06 14:02:02,478 - mmdeploy - INFO - TorchVision: 0.13.0+cu116
2023-02-06 14:02:02,478 - mmdeploy - INFO - OpenCV: 4.7.0
2023-02-06 14:02:02,478 - mmdeploy - INFO - MMCV: 1.7.1
2023-02-06 14:02:02,478 - mmdeploy - INFO - MMCV Compiler: GCC 9.3
2023-02-06 14:02:02,478 - mmdeploy - INFO - MMCV CUDA Compiler: 11.6
2023-02-06 14:02:02,478 - mmdeploy - INFO - MMDeploy: 0.12.0+12b3d18
2023-02-06 14:02:02,478 - mmdeploy - INFO - 

2023-02-06 14:02:02,478 - mmdeploy - INFO - **********Backend information**********
2023-02-06 14:02:02,495 - mmdeploy - INFO - tensorrt:   None
2023-02-06 14:02:02,514 - mmdeploy - INFO - ONNXRuntime:        1.8.1
2023-02-06 14:02:02,514 - mmdeploy - INFO - ONNXRuntime-gpu:    None
2023-02-06 14:02:02,514 - mmdeploy - INFO - ONNXRuntime custom ops:     Available
2023-02-06 14:02:02,516 - mmdeploy - INFO - pplnn:      None
2023-02-06 14:02:02,518 - mmdeploy - INFO - ncnn:       None
2023-02-06 14:02:02,520 - mmdeploy - INFO - snpe:       None
2023-02-06 14:02:02,521 - mmdeploy - INFO - openvino:   None
2023-02-06 14:02:02,522 - mmdeploy - INFO - torchscript:        1.12.0+cu116
2023-02-06 14:02:02,522 - mmdeploy - INFO - torchscript custom ops:     NotAvailable
2023-02-06 14:02:02,542 - mmdeploy - INFO - rknn-toolkit:       None
2023-02-06 14:02:02,542 - mmdeploy - INFO - rknn2-toolkit:      None
2023-02-06 14:02:02,543 - mmdeploy - INFO - ascend:     None
2023-02-06 14:02:02,545 - mmdeploy - INFO - coreml:     None
2023-02-06 14:02:02,546 - mmdeploy - INFO - tvm:        None
2023-02-06 14:02:02,546 - mmdeploy - INFO - 

2023-02-06 14:02:02,546 - mmdeploy - INFO - **********Codebase information**********
2023-02-06 14:02:02,547 - mmdeploy - INFO - mmdet:      2.28.1
2023-02-06 14:02:02,547 - mmdeploy - INFO - mmseg:      None
2023-02-06 14:02:02,547 - mmdeploy - INFO - mmcls:      None
2023-02-06 14:02:02,547 - mmdeploy - INFO - mmocr:      None
2023-02-06 14:02:02,547 - mmdeploy - INFO - mmedit:     None
2023-02-06 14:02:02,547 - mmdeploy - INFO - mmdet3d:    None
2023-02-06 14:02:02,547 - mmdeploy - INFO - mmpose:     None
2023-02-06 14:02:02,547 - mmdeploy - INFO - mmrotate:   None
2023-02-06 14:02:02,547 - mmdeploy - INFO - mmaction:   None
```

**Comment by RunningLeon** [comment]
@shuyuan-wang Hi, for torch==1.12, it's ok to have this Identity ops in onnx. They are optimized by torch that combines multiple Contant Tensor with same values into one.

**Comment by aixiaodewugege** [comment]
@shuyuan-wang Hi, I also try to convert yolox to onnx but I failed with it. Do you change anything in mmdeploy/configs/mmdet/detection/yolox_onnx.py , for I can't convert it using detection_onnxruntime_static.py.

**Comment by shuyuan-wang** [comment]
> @shuyuan-wang Hi, I also try to convert yolox to onnx but I failed with it. Do you change anything in mmdeploy/configs/mmdet/detection/yolox_onnx.py , for I can't convert it using detection_onnxruntime_static.py.

What's your error message?

**Comment by aixiaodewugege** [comment]
> > @shuyuan-wang Hi, I also try to convert yolox to onnx but I failed with it. Do you change anything in mmdeploy/configs/mmdet/detection/yolox_onnx.py , for I can't convert it using detection_onnxruntime_static.py.
> 
> What's your error message?

Here are they. It is very strange. 

(open-mmlab) wushuchen@wushuchen:~/projects/open-mmlab$ python mmdeploy/tools/deploy.py         mmdeploy/configs/mmdet/detection/detection_onnxruntime_dynamic.py         mmdetection/configs/yolox/yolox_tiny_8x8_300e_coco.py         mmdetection/checkpoint/yolox_tiny_8x8_300e_coco_20211124_171234-b4047906.pth         mmdetection/demo/demo.jpg         --work-dir mmdeploy_models/mmdetection/yolox         --device cpu

/home/wushuchen/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/__init__.py:21: UserWarning: On January 1, 2023, MMCV will release v2.0.0, in which it will remove components related to the training process and add a data transformation module. In addition, it will rename the package names mmcv to mmcv-lite and mmcv-full to mmcv. See https://github.com/open-mmlab/mmcv/blob/master/docs/en/compatibility.md for more details.
  'On January 1, 2023, MMCV will release v2.0.0, in which it will remove '
/home/wushuchen/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/__init__.py:21: UserWarning: On January 1, 2023, MMCV will release v2.0.0, in which it will remove components related to the training process and add a data transformation module. In addition, it will rename the package names mmcv to mmcv-lite and mmcv-full to mmcv. See https://github.com/open-mmlab/mmcv/blob/master/docs/en/compatibility.md for more details.
  'On January 1, 2023, MMCV will release v2.0.0, in which it will remove '
/home/wushuchen/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/__init__.py:21: UserWarning: On January 1, 2023, MMCV will release v2.0.0, in which it will remove components related to the training process and add a data transformation module. In addition, it will rename the package names mmcv to mmcv-lite and mmcv-full to mmcv. See https://github.com/open-mmlab/mmcv/blob/master/docs/en/compatibility.md for more details.
  'On January 1, 2023, MMCV will release v2.0.0, in which it will remove '
2023-03-17 20:43:33,257 - mmdeploy - INFO - Start pipeline mmdeploy.apis.pytorch2onnx.torch2onnx in subprocess
load checkpoint from local path: mmdetection/checkpoint/yolox_tiny_8x8_300e_coco_20211124_171234-b4047906.pth
2023-03-17 20:43:33,925 - mmdeploy - WARNING - DeprecationWarning: get_onnx_config will be deprecated in the future. 
2023-03-17 20:43:33,926 - mmdeploy - INFO - Export PyTorch model to ONNX: mmdeploy_models/mmdetection/yolox/end2end.onnx.
/home/wushuchen/projects/open-mmlab/mmdeploy/mmdeploy/core/optimizers/function_marker.py:158: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  ys_shape = tuple(int(s) for s in ys.shape)
/home/wushuchen/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/torch/functional.py:504: UserWarning: torch.meshgrid: in an upcoming release, it will be required to pass the indexing argument. (Triggered internally at ../aten/src/ATen/native/TensorShape.cpp:3190.)
  return _VF.meshgrid(tensors, **kwargs)  # type: ignore[attr-defined]
/home/wushuchen/projects/open-mmlab/mmdeploy/mmdeploy/codebase/mmdet/core/post_processing/bbox_nms.py:97: TracerWarning: torch.tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
  iou_threshold = torch.tensor([iou_threshold], dtype=torch.float32)
/home/wushuchen/projects/open-mmlab/mmdeploy/mmdeploy/codebase/mmdet/core/post_processing/bbox_nms.py:98: TracerWarning: torch.tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
  score_threshold = torch.tensor([score_threshold], dtype=torch.float32)
/home/wushuchen/projects/open-mmlab/mmdeploy/mmdeploy/pytorch/functions/topk.py:28: TracerWarning: torch.tensor results are registered as constants in the trace. You can safely ignore this warning if you use this function to create tensors out of constant variables that would be the same every time you call this function. In any other case, this might cause the trace to be incorrect.
  k = torch.tensor(k, device=input.device, dtype=torch.long)
/home/wushuchen/projects/open-mmlab/mmdeploy/mmdeploy/pytorch/functions/topk.py:34: TracerWarning: Converting a tensor to a Python integer might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  return ctx.origin_func(input, k, dim=dim, largest=largest, sorted=sorted)
/home/wushuchen/projects/open-mmlab/mmdeploy/mmdeploy/mmcv/ops/nms.py:38: TracerWarning: Converting a tensor to a Python float might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  score_threshold = float(score_threshold)
/home/wushuchen/projects/open-mmlab/mmdeploy/mmdeploy/mmcv/ops/nms.py:39: TracerWarning: Converting a tensor to a Python float might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  iou_threshold = float(iou_threshold)
/home/wushuchen/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/ops/nms.py:171: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  assert boxes.size(1) == 4
/home/wushuchen/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/ops/nms.py:172: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  assert boxes.size(0) == scores.size(0)
/home/wushuchen/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/torch/onnx/symbolic_opset9.py:5409: UserWarning: Exporting aten::index operator of advanced indexing in opset 11 is achieved by combination of multiple ONNX operators, including Reshape, Transpose, Concat, and Gather. If indices include negative values, the exported graph will produce incorrect results.
  "Exporting aten::index operator of advanced indexing in opset "
/home/wushuchen/projects/open-mmlab/mmdeploy/mmdeploy/mmcv/ops/nms.py:84: FutureWarning: 'torch.onnx._patch_torch._graph_op' is deprecated in version 1.13 and will be removed in version 1.14. Please note 'g.op()' is to be removed from torch.Graph. Please open a GitHub issue if you need this functionality..
  max_output_boxes_per_class, dtype=torch.long))
/home/wushuchen/projects/open-mmlab/mmdeploy/mmdeploy/mmcv/ops/nms.py:96: FutureWarning: 'torch.onnx._patch_torch._graph_op' is deprecated in version 1.13 and will be removed in version 1.14. Please note 'g.op()' is to be removed from torch.Graph. Please open a GitHub issue if you need this functionality..
  max_output_boxes_per_class, iou_threshold, score_threshold)
2023-03-17 20:43:35,191 - mmdeploy - INFO - Execute onnx optimize passes.
2023-03-17 20:43:35,354 - mmdeploy - INFO - Finish pipeline mmdeploy.apis.pytorch2onnx.torch2onnx
2023-03-17 20:43:35,583 - mmdeploy - INFO - Start pipeline mmdeploy.apis.utils.utils.to_backend in main process
2023-03-17 20:43:35,590 - mmdeploy - INFO - Finish pipeline mmdeploy.apis.utils.utils.to_backend
2023-03-17 20:43:35,590 - mmdeploy - INFO - visualize onnxruntime model start.
/home/wushuchen/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/mmcv/__init__.py:21: UserWarning: On January 1, 2023, MMCV will release v2.0.0, in which it will remove components related to the training process and add a data transformation module. In addition, it will rename the package names mmcv to mmcv-lite and mmcv-full to mmcv. See https://github.com/open-mmlab/mmcv/blob/master/docs/en/compatibility.md for more details.
  'On January 1, 2023, MMCV will release v2.0.0, in which it will remove '
2023-03-17 20:43:37,073 - mmdeploy - INFO - Successfully loaded onnxruntime custom ops from /home/wushuchen/projects/open-mmlab/mmdeploy/mmdeploy/lib/libmmdeploy_onnxruntime_ops.so
corrupted double-linked list
2023-03-17 20:43:37,275 - mmdeploy - ERROR - visualize onnxruntime model failed.
